src/components/SigningModals/MultiScreenModal.js

```diff
@@ -163,6 +168,28 @@ function MultiScreenModal({ visible, screens, onClose }) {
   )
 }
 
+function useScreens(screens) {
+  const { direction, next, prev, step } = useSteps(screens.length)
+  const [screensState, setScreensState] = useState(screens)
+
+  useEffect(() => {
+    setScreensState(screens)
+  }, [screens])
+
+  const getScreen = useCallback(step => screensState[step], [screensState])
+
+  const currentScreen = getScreen(step)
+
+  return {
+    currentScreen,
+    direction,
+    getScreen,
+    next,
+    prev,
+    step,
+  }
+}
```

Comment:
@sohkai – Wrapped this up in a hook per our offline discussion.

Reply:
Small nitpick that we should move the `MultiScreenModal`'s props and defaultProps to be together with the component :).